icu4c/source/i18n/uspoof.cpp

```diff
@@ -697,6 +802,57 @@ uspoof_getSkeleton(const USpoofChecker *sc,
     return destStr.length();
 }
 
+U_CAPI int32_t U_EXPORT2 uspoof_getBidiSkeleton(const USpoofChecker *sc, UBiDiDirection direction,
+                                                const UChar *id, int32_t length, UChar *dest,
+                                                int32_t destCapacity, UErrorCode *status) {
+
+    SpoofImpl::validateThis(sc, *status);
+    if (U_FAILURE(*status)) {
+        return 0;
+    }
+    if (length < -1 || destCapacity < 0 || (destCapacity == 0 && dest != nullptr)) {
+        *status = U_ILLEGAL_ARGUMENT_ERROR;
+        return 0;
+    }
+
+    UnicodeString idStr((length == -1), id, length); // Aliasing constructor
+    UnicodeString destStr;
+    uspoof_getBidiSkeletonUnicodeString(sc, direction, idStr, destStr, status);
+    destStr.extract(dest, destCapacity, *status);
+    return destStr.length();
+}
+
+
+
+U_I18N_API UnicodeString &U_EXPORT2 uspoof_getBidiSkeletonUnicodeString(const USpoofChecker *sc,
+                                                                        UBiDiDirection direction,
+                                                                        const UnicodeString &id,
+                                                                        UnicodeString &dest,
+                                                                        UErrorCode *status) {
+    UBiDi *bidi = ubidi_open();
+    if (direction != UBIDI_LTR && direction != UBIDI_RTL) {
+      *status = U_ILLEGAL_ARGUMENT_ERROR;
+      return dest;
+    }
+    ubidi_setPara(bidi, id.getBuffer(), id.length(), direction,
+                  /*embeddingLevels*/ nullptr, status);
+    if (U_FAILURE(*status)) {
+        return dest;
+    }
+    UnicodeString reordered;
+    std::int32_t const size = ubidi_getProcessedLength(bidi);
+    ubidi_writeReordered(bidi, reordered.getBuffer(size), size,
```

Comment:
reordered.getBuffer(size) could return nullptr if there are no enough memory, in that case, you should set the 
```
*status = U_MEMORY_ALLOCATION_ERROR;
```
and return;